src/context/product/product-context.jsx

```diff
@@ -0,0 +1,31 @@
+import axios from "axios";
+import { createContext, useContext, useState, useEffect } from "react";
+
+const ProductContext = createContext();
+
+const ProductProvider = ({ children }) => {
+    const [products, setProducts] = useState([]);
+
+    useEffect(() => {
+        (async () => {
+            try {
+                const res = await axios.get("/api/products");
+                if (res.status === 200) {
+                    setProducts(res.data.products);
+                }
+            } catch (err) {
+                console.log(err);
```

Comment:
try to give statement in error . It ll help if it throws error .

Just like console.log("Error while getting products",error)